src/lib/fastDownward/fastDownward.ts

```diff
@@ -3,20 +3,27 @@ import { SearchNode } from "./searchNode";
 import { State } from "./state";
 import { HeuristicFunction, CostFunction } from "./fdTypes";
 import { GlobalKnowledge } from "./global-knowledge";
+import { And } from "../ast/and";
+import { Variable } from "../ast/variable";
+import { SkillsRelations } from "../ast/skillsRelation";
 
 /**
  * Compute which LearningUnits are reachable based on the given state.
  */
 function availableActions<LU extends LearningUnit>(
 	currentState: State,
-	learningUnits: ReadonlyArray<LU>
+	learningUnits: ReadonlyArray<LU>,
+	globalKnowledge: GlobalKnowledge
 ) {
 	// Avoid operators that are already in the current state
 	// Ideally we would check that the LearningUnits are not learned twice
 	// However, we can also check that we always learn at least one new skill
+	const skillsRelations = new SkillsRelations(globalKnowledge.skills);
+
 	const usefulLus = learningUnits
 		.filter(unit =>
-			unit.requiredSkills.every(skill => currentState.learnedSkills.includes(skill.id))
+			unit.requiredSkills.evaluate(currentState.learnedSkills, skillsRelations)
```

Comment:
Does this consider `without` expressions?

Reply:
Yes, since the "without" is passed as a paramter to the evaluate function.

Reply:
I dont see the parameter here

Reply:
The "without" parameter is optional to the evaluate function (we did not use it yet).